packages/flutter/lib/src/widgets/app.dart

```diff
@@ -560,6 +560,25 @@ class WidgetsApp extends StatefulWidget {
   /// [supportedLocales] with a matching [Locale.languageCode] is used. If that
   /// fails then the first locale in [supportedLocales] is used. The default
   /// locale resolution algorithm can be overridden with [localeResolutionCallback].
+  ///
+  /// When supporting languages with more than one script, it is recommended
+  /// to specify the scriptCode explicitly. Locales may also be defined without
+  /// countryCode to specify a generic fallback for a particular script. Not
+  /// specifying scriptCode where it is needed (such as Chinese) may lead to
+  /// improperly resolved locales. This may be achieved with the
```

Comment:
"This may be achieved ..." sounds like how to get improperly resolved locales. 

Best to start a new paragraph and explain what the goal of this supportedLocales list is.